Documentation/concepts/observability/hubble-configuration.rst

```diff
@@ -70,3 +64,125 @@ only need to be provided when ``hubble.relay.tls.server.enabled`` is set to
+* Manage certs via a CRD, which is more straightforward than inspecting the PEM file.
+* Auto-renew certificates
+
+**Installation steps**:
+
+1. First install `cert-manager <https://cert-manager.io/docs/installation/>`__ and setup `issuer <https://cert-manager.io/docs/configuration/>`_.
+   Please make sure that your issuer is be able to create certificates under the ``cilium.io`` domain name.
+2. Install/upgrade cilium with bellow configs:
+
+.. code-block:: bash
+
+    --set hubble.tls.auto.enabled=true               # enable automatic TLS certificate generation
+    --set hubble.tls.auto.method=certmanager         # auto generate certificates using cronJob method
+    --set hubble.tls.auto.certValidityDuration=1095  # certificates validity duration in days (default 3 years)
+    --set hubble.tls.auto.certManagerIssuerRef.group="cert-manager.io" # Reference to cert-manager's issuer
+    --set hubble.tls.auto.certManagerIssuerRef.kind="ClusterIssuer"
+    --set hubble.tls.auto.certManagerIssuerRef.name="ca-issuer"
+
+**Troubleshooting**:
+
+If you get the following error while install cilium (and cert-manager), it's because
+cert-manager ValidatingWebhook (which used to verify the Certificate CRD resources)
+is not available (blocked due to CNI is not available).
+
+.. code-block:: bash
+
+    Error: Internal error occurred: failed calling webhook "webhook.cert-manager.io": Post "https://cert-manager-webhook.cert-manager.svc:443/mutate?timeout=10s": dial tcp x.x.x.x:443: connect: connection refused
+
+There are several ways to overcome above issue, such as:
+
+* Option 1: disable validation on cilium installed namespace
+
+.. code-block:: bash
+
+    # We assum cilium installed in kube-system namespace
+    kubectl label namespace kube-system cert-manager.io/disable-validation=true
+
+    helm install cert-manager ...
+    kubectl apply -f issuer.yaml
+    helm install cilium ...
+
+* Option 2: install cert-manager CRDs first
+
+.. code-block:: bash
+
+    # see https://cert-manager.io/docs/installation/helm/#option-1-installing-crds-with-kubectl
+    kubectl create -f cert-manager.crds.yaml
+
+    # cert-manager MUST be installed after cilium
+    helm install cilium ...
+    
+    helm install cert-manager ...
+    kubectl apply -f issuer.yaml
+
+* Option 3: install cert-manager webhook with hostNetwork
+
+.. code-block:: bash
+
+    helm install cert-manager jetstack/cert-manager \
+            --set webhook.hostNetwork=true \
+            --set webhook.tolerations='["operator": "Exists"]'
+    kubectl apply -f issuer.yaml
+
+    helm install cilium ...
+
+* Option 4: upgrade cilium from disabled-TLS installation
+
+.. code-block:: bash
+
+    helm install cilium cilium/cilium \
+            --set hubble.tls.enabled=false \
+            ...
+    
+    helm install cert-manager ...
+    kubectl apply -f issuer.yaml
+
+    # waiting for node ready, and cert-manager available
+
+    helm upgrade cilium ...
```

Comment:
nit: Explicitly use `--set hubble.tls.enabled=true` here to make it clear that we want to enable TLS in this step.